app/src/main/java/com/concordium/wallet/data/walletconnect/ParamsDeserializer.kt

```diff
@@ -45,18 +45,23 @@ class ParamsDeserializer : JsonDeserializer<Params?> {
 
         if (schemaElement.isJsonObject) {
             return try {
-                val schema: Schema = context.deserialize(schemaElement, Schema::class.java)
+                val schema: Schema.ValueSchema =
+                    context.deserialize(schemaElement, Schema.ValueSchema::class.java)
                 if (schema.type == null || schema.value == null) {
                     return null
                 }
                 return schema
             } catch (ex: JsonParseException) {
-                null
+                try {
+                    return context.deserialize(schemaElement, Schema.BrokenSchema::class.java)
+                } catch (ex: JsonParseException) {
+                    return null
```

Comment:
Don’t we in some way want to return an error?

Reply:
Ui error handling was not part of the scope. we just do not show elements that don't parse. other PR adds mechods/chain checks to make sure we only allow hundled actions. and we don't end up with something we cant parse